[ch2035] [`GenericMock`] Add `.addList`

hallais opened the pull request:
feat: Add `GenerickMock.addList`

hallais committed:
[`genericSchema`] Formatting code

hallais committed:
[`GenericMock`] Adding `.addList` [ch2035]

hallais committed:
[`package.json]
Updating `name`
Adding `update-version` and `release-version` commands
Updating libs

hallais committed:
[`CircleCI`] Updating pipeline

ghost commented:
## DeepCode's analysis on [#28d695](https://www.deepcode.ai/app/gh/contartec-team/generic-model-mongoose/6e008d04f9c1d8d42eebb0c7ef548911f89f4795/contartec-team/generic-model-mongoose/28d695860d42a6404cca307584e95bb7a7a9f94d/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=1&) found:

- :information_source: **1** minor issue. :point_down:
## Top issues

<table>
  <thead>
    <tr>
      <th align="left">Description</th>
      <th align="left">Example fixes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td width="77%">The static function getDefaultObject should not be accessed with this. Use the class instance name instead. Occurrences: <ul><li><a href="https://github.com/contartec-team/generic-model-mongoose/blob/28d695860d42a6404cca307584e95bb7a7a9f94d/tests/mocks/models/GenericMock.js#L67">GenericMock.js:67</a></li> <li><a href="https://github.com/contartec-team/generic-model-mongoose/blob/28d695860d42a6404cca307584e95bb7a7a9f94d/tests/mocks/models/GenericMock.js#L85">GenericMock.js:85</a></li> <li><a href="https://github.com/contartec-team/generic-model-mongoose/blob/28d695860d42a6404cca307584e95bb7a7a9f94d/tests/mocks/models/GenericMock.js#L88">GenericMock.js:88</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/contartec-team/generic-model-mongoose/6e008d04f9c1d8d42eebb0c7ef548911f89f4795/contartec-team/generic-model-mongoose/28d695860d42a6404cca307584e95bb7a7a9f94d/pr/_/%2Ftests%2Fmocks%2Fmodels%2FGenericMock.js/javascript%2Fdc_interfile_project%2FStaticAccessThis%2Ftest/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
  </tbody>
</table>


#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/contartec-team/generic-model-mongoose/6e008d04f9c1d8d42eebb0c7ef548911f89f4795/contartec-team/generic-model-mongoose/28d695860d42a6404cca307584e95bb7a7a9f94d/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=1&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=contartec-team)

hallais committed:
[`README`] Adding `doc`

hallais committed:
[`CircleCI`] Fixing unreachable `workflow` path

hallais committed:
[`.npmrc`] Adding conf

hallais committed:
[`CircleCI`] Fixing missing `MONGODB_URL_TEST`

hallais committed:
Removing unnecessary files

hallais committed:
[`GenericModel`]
Formatting code
Refactoring doc

hallais committed:
[`package.json`] Setting package publish to `github's`

hallais committed:
[`CircleCI`] Fixing `working_directory` path

hallais committed:
[`CircleCI`] Adding cache version control through `CACHE_VERSION` env var

hallais committed:
[`CircleCI`] Renaming `save_cache` cache key

hallais committed:
[`package.json`]
Adding `coverage:upload` command
Adding `coverage:upload` call on `test` command

hallais committed:
[`.gitignore`] Adding `package-lock.json`

hallais committed:
[`package.json`] Adding `codecov` lib

contartec committed:
chore(release): updating version to 0.3.0 [skip ci]
